Author: Romain Francoise <rfrancoise@debian.org>
Description:  When iodined runs chrooted it doesn't log anything because
 /dev/log doesn't exist in the chroot and it calls openlog() too late (after
 chrooting). This is upstream ticket #86, and it was fixed by moving the
 openlog() call before chrooting (revisions 914 and 963 of iodined.c),
 resulting in the following patch.
Bug: http://dev.kryo.se/iodine/ticket/86
Bug-Debian: http://bugs.debian.org/573194
Origin: upstream, http://dev.kryo.se/iodine/changeset/914, http://dev.kryo.se/iodine/changeset/963
Reviewed-by: gregor herrmann <gregoa@debian.org>
Last-Update: 2010-03-10

--- a/src/iodined.c
+++ b/src/iodined.c
@@ -1166,6 +1166,13 @@
 	if (foreground == 0) 
 		do_detach();
 	
+#ifdef FREEBSD
+	tzsetwall();
+#endif
+#ifndef WINDOWS32
+	openlog( __progname, LOG_NDELAY, LOG_DAEMON );
+#endif
+
 	if (newroot != NULL)
 		do_chroot(newroot);
 
@@ -1181,9 +1188,6 @@
 #endif
 	}
 
-#ifndef WINDOWS32
-	openlog(__progname, LOG_NOWAIT, LOG_DAEMON);
-#endif
 	syslog(LOG_INFO, "started, listening on port %d", port);
 	
 	tunnel(tun_fd, dnsd_fd, bind_fd);
